docs/melange-book/counter-1.md

```diff
@@ -0,0 +1,127 @@
+# Counter, Part 1
+
+We're going build the classic starter app, the counter, using
+[ReasonReact](https://reasonml.github.io/reason-react/). Let's clone the
+tutorial template and initialize it:
+
+    git clone https://github.com/melange-re/melange-tutorial
+    cd melange-tutorial
+    make init
+
+To start webpack, run `make serve`. The app will be served at
+http://localhost:8081/. In another terminal window, start the Reason compiler in
+watch mode by running `make watch`.
+
+Open `Index.re` and you'll see this:
+
+```reason
+module App = {
+  [@react.component]
+  let make = () => <div> {React.string("welcome to my app")} </div>;
+};
+```
+
+This is just about the simplest component you can make. Unlike in normal React,
+we must wrap strings inside function calls to convert them to the
+`React.element` type. This is exactly what the `React.string` function does--if
+you hover over it, you'll see that it displays the type `string =>
+React.element`. The `make` function itself has the type `unit => React.element`,
+meaning it takes no arguments and returns an object of type `React.element`.
```

Comment:
Technically, the function takes one argument of type `unit`: https://www2.lib.uchicago.edu/keith/ocaml-class/data.html#unit. All functions in OCaml take precisely 1 argument. (This kind of leads directly into currying, I'm not sure if it's worth introducing it here).

Reply:
Corrected the inaccuracy, but didn't get into currying :sweat_smile: